twine/settings.py

```diff
@@ -253,12 +253,20 @@ def _handle_authentication(self, username, password):
             username,
             self.repository_config
         )
-        self.password = utils.get_password(
-            self.repository_config['repository'],
-            self.username,
-            password,
-            self.repository_config,
-        )
+        if self.client_cert:
+            self.password = utils.get_password_no_prompt(
```

Comment:
Good instincts not passing a boolean parameter but instead selecting a different function.